controllers/email.js

```diff
@@ -94,58 +49,85 @@ const getTransport = async () => {
   }
 };
 
+// cb is callback, cb(err, null) means if err, get err, else null
+// IF WE WANT TO CHANGE THE RECIPIENT ADDRESS LATER, MUST AUTHORIZE IN MAILGUN SYSTEM FIRST
+const sendMail = async (from, to, subject, message, attachments = undefined) => {
+  console.log('SENDMAIL');
+  try {
+    const transporter = await getTransport();
+
+    if (transporter) {
+      const mailOptions = {
+        from,
+        to,
+        subject,
+        html: message,
+        attachments,
+      };
+
+      const data = await transporter.sendMail(mailOptions);
+
+      console.log(data);
+      if (!data) {
+        return { success: false };
+      }
+      if (process.env.NODE_ENV === 'development') {
+        console.log('Preview URL: %s', nodemailer.getTestMessageUrl(data));
+        return { success: true, data: nodemailer.getTestMessageUrl(data) };
+      }
+      return { success: true, data: '' };
+    }
+    return { success: false };
+  } catch (e) {
+    console.log(e);
+  }
+};
+
 const sendVerificationEmail = async (to, hash) => {
   console.log('MAIL');
   const attachments = [
     {
       filename: 'instagram2x.png',
       path: path.resolve(__dirname, '../resources/email/instagram2x.png'),
-      cid: 'instagram2x.png', //same cid value as in the html img src
+      cid: 'instagram2x.png', // same cid value as in the html img src
     },
     {
       filename: 'website2x.png',
       path: path.resolve(__dirname, '../resources/email/website2x.png'),
-      cid: 'website2x.png', //same cid value as in the html img src
+      cid: 'website2x.png', // same cid value as in the html img src
     },
     {
       filename: 'mail2x.png',
       path: path.resolve(__dirname, '../resources/email/mail2x.png'),
-      cid: 'mail2x.png', //same cid value as in the html img src
+      cid: 'mail2x.png', // same cid value as in the html img src
     },
     {
       filename: 'new-donate-gifts-logo-2.png',
-      path: path.resolve(
-        __dirname,
-        '../resources/email/new-donate-gifts-logo-2.png'
-      ),
-      cid: 'new-donate-gifts-logo-2.png', //same cid value as in the html img src
+      path: path.resolve(__dirname, '../resources/email/new-donate-gifts-logo-2.png'),
+      cid: 'new-donate-gifts-logo-2.png', // same cid value as in the html img src
     },
     {
       filename: 'Img1_2x.jpg',
       path: path.resolve(__dirname, '../resources/email/Img1_2x.jpg'),
-      cid: 'Img1_2x.jpg', //same cid value as in the html img src
+      cid: 'Img1_2x.jpg', // same cid value as in the html img src
     },
   ];
 
-  const body = template.replace(
-    'linkplaceholder',
-    `${process.env.BASE_URL}/users/verify/${hash}`
-  );
+  const body = template.replace('linkplaceholder', `${process.env.BASE_URL}/users/verify/${hash}`);
 
   console.log('BODY');
-  return await sendMail(
+  return sendMail(
```

Comment:
wont this return a promise now? not sure if that will cause issues in the signup method